### DISCUSSION TITLE
How to use mdast-util-from-markdown to support Chinese and emphasis when parsing

### DISCUSSION THREAD

**Discussion by rxliuli** [micromark/micromark#149] [Q&A] [answered]
As in the title, the current standard markdown cannot parse content that contains Chinese and emphasis at the same time.

For example `**真没想到我这么快就要死了，**她有些自暴自弃地想着。`, will get `<p>**真没想到我这么快就要死了，**她有些自暴自弃地想着。</p>`, expect to get `<p><strong>真没想到我这么快就要死了，</strong>她有些自暴自弃地想着。</p>`

There is a PR about this issue open in cmark for 6 years and still not resolved, ref: <https://github.com/commonmark/cmark/pull/208>.
I'd like to know how to solve this problem myself directly through the plugin. I read some readme and it seems the entry is mdastExtension or micromark extension, but I'm still not quite sure what to do. Asking GPT-4 did not yield meaningful examples.

**Comment by wooorm** [comment]
Hi. Our projects follow CommonMark: standard markdown. Particularly around parsing.

So, a) if possible you should discuss on the issue you found.

b) can you use `**真**，她。` instead of `**真，**她。`?:

```markdown
**真，**她。

**真**，她。
```

**真，**她。

**真**，她。


and c) you can use `&ZeroWidthSpace;` to force punctuation around the characters.
Showing all possible examples:

```markdown
1. &ZeroWidthSpace;**真，**她。
2. **&ZeroWidthSpace;真，**她。
3. **真，&ZeroWidthSpace;**她。
4. **真，**&ZeroWidthSpace;她。
5. &ZeroWidthSpace;**&ZeroWidthSpace;真，**她。
6. &ZeroWidthSpace;**真，&ZeroWidthSpace;**她。
7. &ZeroWidthSpace;**真，**&ZeroWidthSpace;她。
8. **&ZeroWidthSpace;真，&ZeroWidthSpace;**她。
9. **&ZeroWidthSpace;真，**&ZeroWidthSpace;她。
10. **真，&ZeroWidthSpace;**&ZeroWidthSpace;她。
11. &ZeroWidthSpace;**&ZeroWidthSpace;真，&ZeroWidthSpace;**她。
12. &ZeroWidthSpace;**&ZeroWidthSpace;真，**&ZeroWidthSpace;她。
13. &ZeroWidthSpace;**&ZeroWidthSpace;真，&ZeroWidthSpace;**&ZeroWidthSpace;她。
14. &ZeroWidthSpace;**真，&ZeroWidthSpace;**&ZeroWidthSpace;她。
15. **&ZeroWidthSpace;真，&ZeroWidthSpace;**&ZeroWidthSpace;她。
```

1. &ZeroWidthSpace;**真，**她。
2. **&ZeroWidthSpace;真，**她。
3. **真，&ZeroWidthSpace;**她。
4. **真，**&ZeroWidthSpace;她。
5. &ZeroWidthSpace;**&ZeroWidthSpace;真，**她。
6. &ZeroWidthSpace;**真，&ZeroWidthSpace;**她。
7. &ZeroWidthSpace;**真，**&ZeroWidthSpace;她。
8. **&ZeroWidthSpace;真，&ZeroWidthSpace;**她。
9. **&ZeroWidthSpace;真，**&ZeroWidthSpace;她。
10. **真，&ZeroWidthSpace;**&ZeroWidthSpace;她。
11. &ZeroWidthSpace;**&ZeroWidthSpace;真，&ZeroWidthSpace;**她。
12. &ZeroWidthSpace;**&ZeroWidthSpace;真，**&ZeroWidthSpace;她。
13. &ZeroWidthSpace;**&ZeroWidthSpace;真，&ZeroWidthSpace;**&ZeroWidthSpace;她。
14. &ZeroWidthSpace;**真，&ZeroWidthSpace;**&ZeroWidthSpace;她。
15. **&ZeroWidthSpace;真，&ZeroWidthSpace;**&ZeroWidthSpace;她。

Looking at those, we see that 4, 7, 9, 10, 12, 13, 14, 15. All because of the asterisks you want to use as *closing* the emphasis, which used to have a punctuation before and not after them, and now has punctuation before *and* after them.

**Reply by rxliuli** [reply]
I'm currently using `**真，** 她。`, And remove extra whitespace via plugin when rendering to html, but this is not very friendly for other documentation maintainers, so I hope there is a way to directly support `**真，**她。`.
Also, I don't intend to discuss whether to modify the markdown specification, because its modification process is very slow. I'm just wondering if it's possible to render it properly with a plugin, and if so, how.

**Reply by wooorm** [reply, marked as answer]
> I'm just wondering if it's possible to render it properly with a plugin, and if so, how.

micromark is extendable. But a) you need to be a very good programmer, b) it’s going to get very complex, c) you shouldn’t do it because it isn’t markdown anymore.
Please read the *entire* “Extensions” chapter first: https://github.com/micromark/micromark#extensions.

>  I don't intend to discuss whether to modify the markdown specification, because its modification process is very slow.

I hope you can do so: if that issue can be solved, it helps many speakers of Chinese (and speakers of other East Asian languages)!
